<launch>
    <!-- Define parameters for recorder -->
    <param name="outputDirectory" type="string" value="$(env HOME)/Documents/university/specialCourse/recordedData/" />
    <param name="outputFileName" type="string" value="zed_" />
    <param name="topicName" type="string" value="/zed2/zed_node/point_cloud/cloud_registered" />
    <param name="numberOfMessages" type="int" value="3"/>

    <!-- Start the Zed 2 SDK -->
    <include file="$(find zed_wrapper)/launch/zed2.launch"/>

    <!-- Record data and save to PCD file -->
    <node name="pcdRecorder" pkg="special_course" type="pcdRecorder"  output="screen"/>

    <!-- Start Viewer if rviz in args -->
	<arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find special_course)/rviz_cfg/zed2.rviz" />
    </group>
</launch>
